### PR TITLE
Test policy in pr

### PR DESCRIPTION
**Pull request by chrisns**
## Related issue
n/a

## Milestone of this PR
n/a
## What type of PR is this

> /kind feature

## Proposed Changes

Use github actions to test the policies (also use dependabot to keep actions up to date)

Note this did reveal that one of the policies doesn't pass against the current release cli, it does pass in v1.5.2-rc1 though.
I could make this pull the latest pre-release if one exists, but that feels a bit fragile, ideally the policies repo should match what the current release can do?

### Proof Manifests

n/a

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] My PR contains new or altered behavior to Kyverno and
  
  - [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.

## Further Comments

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (realshuting) [inline .github/workflows/ci.yml:9]: Can we also trigger tests for `release*` branches?

**Review by realshuting — COMMENTED** [review]
/lgtm

Added a minor comment.

**Comment by chipzoller** [comment]
> Note this did reveal that one of the policies doesn't pass against the current release cli, it does pass in v1.5.2-rc1 though.
> I could make this pull the latest pre-release if one exists, but that feels a bit fragile, ideally the policies repo should match what the current release can do?

Yes, this was our mistake. Going forward this won't happen.

**Comment by chrisns** [comment]
> Yes, this was our mistake. Going forward this won't happen.

not after this PR is merged 😉

**Review by realshuting — APPROVED** [review]

**Comment by realshuting** [comment]
Can we merge this PR? @chipzoller

**Comment by chipzoller** [comment]
Yes, looks good.

**Comment by chrisns** [comment]
do we know how far off the fix in upstream kyverno will resolve the failing tests? https://github.com/kyverno/policies/runs/4408313139?check_suite_focus=true#step:4:360

**Comment by chipzoller** [comment]
The [corresponding PR](https://github.com/kyverno/kyverno/pull/2569) was tagged for 1.5.2 (for which RCs are now available), but I don't see it flagged in the notes.

**Comment by chrisns** [comment]
um :( what would you rather:

1. temporarily use a pre-release kyverno in ci (two prs, one to put it in, another to remove it, tests will let us know when we can merge)
2. remove the failing policies, or skip the tests (again two prs)
3. just wait for the upstream release

**Comment by chipzoller** [comment]
I think for now I'd rename the test manifest and leave it intact (and create a new issue reminding us we need to switch the name back once that feature is available so we don't forget) which will cause the `kyverno test` command to simply skip over it.

**Comment by chrisns** [comment]
cool @chipzoller i'll do that now, the second pr will take care of the reminding issue bit and let us know when we can :)

**Pull request merged**
